### PR TITLE
Gigatron: a little more meat on the bones

### PR DESCRIPTION
**Pull request by Sterophonick**
I hooked up the keypad to the CPU, and fixed a plumbing issue with the outx callback.

### CODE REVIEW

**Comment by Sterophonick** [comment]
Why does the Visual Studio build fail? I didn't even touch emumem.

File: src/devices/cpu/gigatron/gigatron.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #include "gigatrondasm.h"
 
 
-DEFINE_DEVICE_TYPE(GTRON, gigatron_cpu_device, "gigatron_cpu", "Gigatron CPU")
+DEFINE_DEVICE_TYPE(GTRON, gigatron_cpu_device, "gigatron_cpu", "Gigatron")
```
Comment (cuavas) [inline src/devices/cpu/gigatron/gigatron.cpp:16]: Please stop doing this.  “Gigatron” is the name of the CPU, the instruction set, and the computer.  It needs to say “CPU” here to be unambiguous.

Comment (Sterophonick) [reply src/devices/cpu/gigatron/gigatron.cpp:16]: Alright, makes sense.

File: src/mame/drivers/gigatron.cpp [hunk 2]
```diff
@@ -82,6 +83,11 @@ void gigatron_state::blinkenlights(uint8_t data)
 	lights_changed ^= light;
 }
 
+uint8_t gigatron_state::inputs()
+{
+	return m_io_inputs->read() ^ 0xFF;
+}
+
```
Comment (cuavas) [inline src/mame/drivers/gigatron.cpp:90]: You could avoid this trampoline entirely either by using active high inputs in the input port definitions, or by using something like `m_maincpu->ir_cb().set_ioport("GAMEPAD").invert();` in your machine configuration.

**Comment by Sterophonick** [comment]
Alright, thanks for telling me that!

**Comment by cuavas** [comment]
Also, how is the keypad “hooked up” when the CPU never actually uses `m_ir_cb` at all?

**Comment by Sterophonick** [comment]
How can I make m_ir_cb set any value given to it to inReg?

**Comment by Sterophonick** [comment]
Would it be like this?
m_inReg = m_ir_cb(0, 0xFF);

**Comment by cuavas** [comment]
How about you actually ensure stuff compiles, links, validates, and at least does something before opening pull requests?

**Comment by Sterophonick** [comment]
That's fair.

**Pull request merged**